librz/arch/isa/msp430/msp430_disas.h

```diff
@@ -43,23 +42,44 @@ enum msp430_twoop_opcodes {
 	MSP430_BIS,
 	MSP430_XOR,
 	MSP430_AND,
-};
+} Msp430TwoopOpcodes;
 
-enum msp430_addr_modes {
-	MSP430_DIRECT,
-	MSP430_INDEXED,
-	MSP430_INDIRECT,
-	MSP430_INDIRECT_INC,
-};
+typedef enum msp430_emulated_opcodes {
+	MSP430_ADC,
+	MSP430_BR,
+	MSP430_CLR,
+	MSP430_CLRC,
+	MSP430_CLRN,
+	MSP430_CLRZ,
+	MSP430_DADC,
+	MSP430_DEC,
+	MSP430_DECD,
+	MSP430_DINT,
+	MSP430_EINT,
+	MSP430_INC,
+	MSP430_INCD,
+	MSP430_INV,
+	MSP430_NOP,
+	MSP430_POP,
+	MSP430_RET,
+	MSP430_RLA,
+	MSP430_RLC,
+	MSP430_SBC,
+	MSP430_SETC,
+	MSP430_SETN,
+	MSP430_SETZ,
+	MSP430_TST
+} Msp430EmulatedOpcodes;
 
-enum msp430_cmd_type {
+typedef enum msp430_cmd_type {
 	MSP430_ONEOP,
 	MSP430_TWOOP,
 	MSP430_JUMP,
-	MSP430_INV,
-};
+	MSP430_EMULATE,
+	MSP430_INVALID,
```

Comment:
Careful with the `INVALID` at the last position.
If some struct is initialized with all `0`, the value won't be `INVALId`. Is this on purpose?

Reply:
Hmm, I actually don't know, I'm not the one who wrote the disassembler. Porbably the original writer just didn't care enough.

From what I have seen the code do, that's a non-problem. Nobody relies on the fact that an uninitialized command struct is invalid, the entrypoint of the diassembler explicitly sets it on every control path.